src/hw/connectApp.js

```diff
@@ -222,6 +231,18 @@ const cmd = ({
               return of(e);
             }
           }),
+          retryWhen((errors: Observable<*>) =>
+            errors.pipe(
+              filter(
+                (e) =>
+                  !!e &&
+                  e instanceof TransportStatusError &&
+                  e.statusCode === 0x6982 // device is locked in a special case
+              ),
+              tap(() => o.next({ type: "unresponsiveDevice" })),
+              delay(5000)
+            )
```

Comment:
I feel using retryWhen is dangerous here, specifically as connectApp is supposed to be one step of the bigger loop which is actions/app.js

but i miss a bit to understand the big picture here.

to me the fix should not happen here but more on the connectApp to returns something like 

```
          return of({ type: "unresponsiveDevice" });
```

at the place where it used to return "ask-open-app"

Reply:
i tried that but it just moves the blocking issue elsewhere.
Since we unlock the device on an app opened we dont get the loop restart from a device change we just get stuck on the "unlock device" (which just replaces the "open app" from before)
The only way to restart the job is then to quit the app and then the loop restarts
maybe i haven't checked too much how other apps behave in that case :thinking: